pandas/core/series.py

```diff
@@ -3783,19 +3783,10 @@ def argsort(
         mask = isna(values)
 
         if mask.any():
-            # TODO(3.0): once this deprecation is enforced we can call
-            #  self.array.argsort directly, which will close GH#43840 and
-            #  GH#12694
-            warnings.warn(
-                "The behavior of Series.argsort in the presence of NA values is "
-                "deprecated. In a future version, NA values will be ordered "
-                "last instead of set to -1.",
-                FutureWarning,
-                stacklevel=find_stack_level(),
+            raise TypeError(
```

Comment:
Based on the `TODO` comment, it appears this should dispatch to `self.array.argsort`

Also could you see if it closes those associated issues referenced in the comment?

Reply:
> Based on the `TODO` comment, it appears this should dispatch to `self.array.argsort`
> 
sorry, it’s my mistake, I dispatched `Series.argsort` to `EA.argsort`.
> Also could you see if it closes those associated issues referenced in the comment?
>
Yes, it will close issues referenced in the comment.

@mroeschke could you please take a look at this PR? I think CI failures are unrelated to my changes.

Reply:
cc @jbrockmendel for a double check